src/ja/pixivkomikku/src/eu/kanade/tachiyomi/extension/ja/pixivkomikku/PixivKomikkuUtil.kt

```diff
@@ -0,0 +1,246 @@
+    val hexadecimalTimeHash = saltedTimeHash.joinToString("") {
+        var hex = Integer.toHexString(it.toInt())
+        if (hex.length < 2) {
+            hex = "0$hex"
+        }
+        return@joinToString hex
+    }
+
+    return Pair(timeFormatted, hexadecimalTimeHash)
+}
+
+private fun getCurrentTimeZoneOffsetString(): String {
+    val timeZone = TimeZone.getDefault()
+    val offsetInMillis = timeZone.rawOffset
+
+    val hours = offsetInMillis / (1000 * 60 * 60)
+    val minutes = (offsetInMillis % (1000 * 60 * 60)) / (1000 * 60)
+
+    val sign = if (hours >= 0) "+" else "-"
+    val formattedHours = String.format("%02d", abs(hours))
+    val formattedMinutes = String.format("%02d", abs(minutes))
+
+    return "$sign$formattedHours:$formattedMinutes"
+}
+
+@OptIn(ExperimentalUnsignedTypes::class)
+internal fun ResponseBody.toBase64ImageString(key: String): String {
+    // get the image color data
+    val bytes = this.bytes()
+    val shuffledImageBitmap = BitmapFactory.decodeByteArray(bytes, 0, bytes.size)
+
+    val width = shuffledImageBitmap.width
+    val height = shuffledImageBitmap.height
+
+    val shuffledImageArray = UByteArray(width * height * BYTES_PER_PIXEL)
+
+    var index = 0
+    for (y in 0 until shuffledImageBitmap.height) {
+        for (x in 0 until shuffledImageBitmap.width) {
+            val pixel = shuffledImageBitmap.getPixel(x, y)
+
+            val alpha = pixel shr 24 and 0xff
+            val red = pixel shr 16 and 0xff
+            val green = pixel shr 8 and 0xff
+            val blue = pixel and 0xff
+
+            shuffledImageArray[index++] = alpha.toUByte()
+            shuffledImageArray[index++] = red.toUByte()
+            shuffledImageArray[index++] = green.toUByte()
+            shuffledImageArray[index++] = blue.toUByte()
+        }
+    }
+
+    // deShuffle the shuffled image
+    val deShuffledImageArray = deShuffleImage(shuffledImageArray, width, height, key)
+
+    // place it back together
+    val deShuffledImageBitmap = Bitmap.createBitmap(width, height, Bitmap.Config.ARGB_8888)
+    val canvas = Canvas(deShuffledImageBitmap)
+
+    index = 0
+    for (y in 0 until deShuffledImageBitmap.height) {
+        for (x in 0 until deShuffledImageBitmap.width) {
+            // it's rgba
+            val red = deShuffledImageArray[index++]
+            val green = deShuffledImageArray[index++]
+            val blue = deShuffledImageArray[index++]
+            val alpha = deShuffledImageArray[index++]
+
+            canvas.drawPoint(
+                x.toFloat(),
+                y.toFloat(),
+                Paint().apply {
+                    setARGB(red.toInt(), green.toInt(), blue.toInt(), alpha.toInt())
+                },
+            )
+        }
+    }
+
+    val outStream = ByteArrayOutputStream()
+    deShuffledImageBitmap.compress(Bitmap.CompressFormat.PNG, 100, outStream)
+
+    return Base64.encodeToString(outStream.toByteArray(), Base64.DEFAULT)
```

Comment:
you should make an interceptor to descramble image response directly instead of doing base64 data image for `DataImageInterceptor`